torchvision/datasets/dtd.py

```diff
@@ -0,0 +1,86 @@
+import os
+import pathlib
+from typing import Optional, Callable, Union
+
+import PIL.Image
+
+from .utils import verify_str_arg, download_and_extract_archive
+from .vision import VisionDataset
+
+
+class DTD(VisionDataset):
+    """`Describable Textures Dataset (DTD) <https://www.robots.ox.ac.uk/~vgg/data/dtd/>`_.
+
+    Args:
+        root (string): Root directory of the dataset.
+        split (string, optional): The dataset split, supports ``"train"`` (default), ``"val"``, or ``"test"``.
+        fold (string or int, optional): The dataset fold. Should be ``1 <= fold <= 10``. Defaults to ``1``.
+        transform (callable, optional): A function/transform that  takes in an PIL image and returns a transformed
+            version. E.g, ``transforms.RandomCrop``.
+        target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+    """
+
+    _URL = "https://www.robots.ox.ac.uk/~vgg/data/dtd/download/dtd-r1.0.1.tar.gz"
+    _MD5 = "fff73e5086ae6bdbea199a49dfb8a4c1"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = "train",
+        fold: Union[str, int] = 1,
```

Comment:
The 10 splits seem to be rather specific to the paper. It's great to have this parameter for reproducibility, but I wonder if we should allow an "all" option here, and perhaps it could be the default?

Reply:
IIUC, each fold contains all images across the splits. So if we want all images, we should adapt the `split` parameter.